finagle-core/src/main/scala/com/twitter/finagle/tracing/TraceInitializerFilter.scala

```diff
@@ -229,25 +229,41 @@ object ClientTracingFilter {
 private[finagle] object WireTracingFilter {
   val role = Stack.Role("WireTracingFilter")
 
-  case class TracingFilter[Req, Rep](
-    label: String,
-    finagleVersion: () => String = () => Init.finagleVersion
-  ) extends AnnotatingTracingFilter[Req, Rep](
-    label,
-    "clnt",
-    Annotation.WireSend,
-    Annotation.WireRecv,
-    Annotation.WireRecvError(_),
-    finagleVersion,
-    false)
-
-  def module[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] =
+  def clientModule[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] =
     new Stack.Module1[param.Label, ServiceFactory[Req, Rep]] {
-      val role = ClientTracingFilter.role
+      val role = WireTracingFilter.role
       val description = "Report finagle information and wire send/recv events"
       def make(_label: param.Label, next: ServiceFactory[Req, Rep]) = {
         val param.Label(label) = _label
-        TracingFilter[Req, Rep](label) andThen next
+        Filter(label) andThen next
+      }
+
+      case class Filter(
+        label: String,
+        finagleVersion: () => String = () => Init.finagleVersion
+      ) extends AnnotatingTracingFilter[Req, Rep](
+        label,
+        "clnt",
+        Annotation.WireSend,
+        Annotation.WireRecv,
+        Annotation.WireRecvError(_),
+        finagleVersion,
+        false)
+    }
+
+  def serverModule[Req, Rep]: Stackable[ServiceFactory[Req, Rep]] =
+    new Stack.Module0[ServiceFactory[Req, Rep]] {
+      val role = WireTracingFilter.role
+      val description = "Report wire recv/send events"
+      def make(next: ServiceFactory[Req, Rep]) = filter andThen next
+
+      object filter extends Filter[Req, Rep, Req, Rep] {
+        private[this] def recv(): Unit = Trace.record(Annotation.WireRecv)
+        private[this] def send(rsp: Rep): Unit = Trace.record(Annotation.WireSend)
+        def apply(req: Req, service: Service[Req, Rep]) = {
+          recv()
+          service(req).onSuccess(send)
```

Comment:
so unfortunately this doesn't do the trick because the method will be eta-expanded into a function value on each request and that costs an allocation. anything but a function value burns us. I'll fix it up locally if that's cool with you.